src/equations/kdv_1d.jl

```diff
@@ -0,0 +1,175 @@
+    Dt = Differential(t)
+    Dx = Differential(x)
+
+    @variables g D pi
+
+    η = 1 + exp(-t // 2) * sin(pi*(2 * (x - t // 2)))
+    η_t = (Dt(η))
+    η_x = (Dx(η))
+    η_xxx = (Dx(Dx(Dx(η))))
+
+    source_kdv = η_t + sqrt(g * D) * η_x + 3//2 * sqrt(g / D) * η * η_x + 1//6 * sqrt(g * D) * D^2 * η_xxx
+
+    source_julia_form = simplify(expand_derivatives(source_kdv))
+``` 
+"""
+function source_terms_manufactured(q, x, t, equations::KdVEquation1D)
+    g = gravity(equations)
+    D = equations.D
+
+    a1 = sinpi(2x - t)
+    a2 = cospi(2x - t)
+    b1 = exp(-t / 2)
+    c0 = sqrt(g * D)
+    c1 = sqrt(g / D)
+
+    dη = -0.5 * a1 * b1 - pi * a2 * b1 +
+         2pi * a2 * c0 * b1 +
+         3pi * a2 * c1 * (1 + a1 * b1) * b1 -
+         (4 / 3) * D^2 * pi^3 * a2 * c0 * b1
+
+    return SVector(dη)
+end
+
+function create_cache(mesh, equations::KdVEquation1D,
+                      solver, initial_condition,
+                      ::BoundaryConditionPeriodic,
+                      RealT, uEltype)
+    g = gravity(equations)
+    D = equations.D
+    DD = D^2
+    c_0 = sqrt(g * D)
+    c_1 = 0.5 * sqrt(g / D)
+
+    # We use `DiffCache` from PreallocationTools.jl to enable automatic/algorithmic differentiation
+    # via ForwardDiff.jl. 
+    # 1: eta
+    N = ForwardDiff.pickchunksize(1 * nnodes(mesh))
+    template = ones(RealT, nnodes(mesh))
+
+    eta2 = DiffCache(zero(template), N)
+    eta2_x = DiffCache(zero(template), N)
+
+    eta_x = DiffCache(zero(template), N)
+    eta_xxx = DiffCache(zero(template), N)
+
+    if solver.D1 isa PeriodicUpwindOperators
+        D1 = solver.D1.central
+
+        # calculate the third derivative operator using upwind operators
+        D3 = sparse(solver.D1.plus) * sparse(solver.D1.minus) * sparse(solver.D1.central)
+
+    else
+        D1 = solver.D1
+        D3 = solver.D3
+    end
+
+    cache = (; D1, D3, c_0, c_1, DD, eta2, eta2_x, eta_x, eta_xxx,)
+    return cache
+end
+
+function rhs!(dq, q, t, mesh, equations::KdVEquation1D, initial_condition,
+              ::BoundaryConditionPeriodic, source_terms, solver, cache)
+    eta, = q.x
+    deta, = dq.x
+
+    (; D1, D3, c_0, c_1, DD,) = cache
+    # In order to use automatic differentiation, we need to extract
+    # the storage vectors using `get_tmp` from PreallocationTools.jl
+    # so they can also hold dual numbers when needed.
+    eta2 = get_tmp(cache.eta2, eta)
+    eta2_x = get_tmp(cache.eta2_x, eta2)
+    eta_x = get_tmp(cache.eta_x, eta)
+    eta_xxx = get_tmp(cache.eta_xxx, eta)
+
+    @trixi_timeit timer() "hyperbolic" begin
+        @.. eta2 = eta^2
+        # eta2_x = D1 * eta2
+        mul!(eta2_x, D1, eta2)
+
+        # eta_x = D1 * eta
+        mul!(eta_x, D1, eta)
+
+        # eta_xxx = D3 * eta
+        mul!(eta_xxx, D3, eta)
+
+        @.. deta = -1.0 * (c_0 * eta_x +
+                    c_1 * eta * eta_x +
+                    c_1 * eta2_x +
+                    1 / 6 * c_0 * DD * eta_xxx)
+    end
```

Comment:
```suggestion

        @.. deta = -1.0 * (c_0 * eta_x +
                    c_1 * (eta * eta_x +
                           eta2_x)
    end
        
    @trixi_timeit timer() "third-order derivatives" begin
        # eta_xxx = D3 * eta
        mul!(eta_xxx, D3, eta)
        
        @.. deta -= 1 / 6 * c_0 * DD * eta_xxx
    end
```
The third order derivative terms are not hyperbolic. So I would put them into a separate timer.